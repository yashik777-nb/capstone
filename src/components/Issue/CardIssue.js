import React from "react";
import { withRouter } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

const styles = (theme) => ({
  root: {
    minWidth: 300,
    maxWidth: 300,
    margin: 5,
    display: "inline-block",
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  issueDescription: {
    wordWrap: `break-word`,
  },
});

class CardIssue extends React.Component {
  issueClicked(e) {
    if (this.props.authenticated) {
      if (window.confirm("Are you sure you want to view the details ?"))
        this.props.history.push(`/issues/${this.props.issueDescription}`);
    } else {
      if (window.confirm("Please Sign In/Register to View More Details"))
        this.props.history.push(`/signin`);
    }
    // if (window.confirm("Are you sure you want to view the details ?"))
    //   this.props.history.push(`/issues/${this.props.issueDescription}`);
    // this.props.history.push(`/signin`);
  }

  render() {
    const { classes } = this.props;
    return (
      <Card className={classes.root} variant="outlined">
        <CardContent>
          <Typography
            className={classes.title}
            color="textSecondary"
            gutterBottom
          >
            {this.props.title}
          </Typography>
          <Typography
            variant="h5"
            component="h2"
            className={classes.issueDescription}
          >
            {this.props.issueDescription}
          </Typography>
          {this.props.customize.severity ? (
            <Typography className={classes.pos} color="textSecondary">
              Severity: {this.props.severity}
            </Typography>
          ) : null}
          {this.props.customize.status ? (
            <Typography className={classes.pos} color="textSecondary">
              Status: {this.props.status}
            </Typography>
          ) : null}
          {this.props.customize.createdDate ? (
            <Typography variant="body2" component="p">
              Created Date: {this.props.createdDate}
            </Typography>
          ) : null}
          {this.props.customize.resolvedDate ? (
            <Typography variant="body2" component="p">
              Resolved Date: {this.props.resolvedDate}
            </Typography>
          ) : null}
        </CardContent>
        <CardActions>
          <Button size="small" onClick={(e) => this.issueClicked(e)}>
            View Details
          </Button>
        </CardActions>
      </Card>
    );
  }
}

export default withStyles(styles, { withTheme: true })(withRouter(CardIssue));
