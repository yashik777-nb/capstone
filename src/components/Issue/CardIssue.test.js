import React from "react";
import { configure } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import Typography from "@material-ui/core/Typography";

import CardIssue from "./CardIssue";

configure({ adapter: new Adapter() });

const styles = {
  root: "",
  title: "",
  pos: "",
  issueDescription: "",
};

const props = {
  classes: styles,
  customize: {
    severity: true,
    status: true,
    createdDate: true,
    resolvedDate: true,
  },
  id: 1,
  title: "Issue One",
  issueDescription: "Issue Description One",
  severity: "Critical",
  status: "Open",
  createdDate: "28/02/2021  ",
  resolvedDate: "29/03/2021",
  views: 5,
  backGroundColor: "#7828BF",
  hoverBackGroundColor: "#9C5BB7",
};

describe("About Page Tests", () => {
  let wrapper;

  beforeEach(() => {
    const render = createMount();
    wrapper = render(
      <BrowserRouter>
        <CardIssue {...props} />
      </BrowserRouter>
    );
  });

  it("Should render six Typography", () => {
    expect(wrapper.find(Typography).length).toEqual(6);
  });

  test("Should Render a Card Issue", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
