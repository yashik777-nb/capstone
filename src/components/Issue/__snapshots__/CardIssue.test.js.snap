// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page Tests Should Render a Card Issue 1`] = `ReactWrapper {}`;
